py3status/modules/imap.py

```diff
@@ -65,61 +75,98 @@ class Meta:
 
-        if mail_count == 0 and self.hide_if_zero:
+        if self.mail_count == 0 and self.hide_if_zero:
             response['full_text'] = ''
         else:
-            response['full_text'] = self.py3.safe_format(self.format, {'unseen': mail_count})
+            response['full_text'] = self.py3.safe_format(self.format, {'unseen': self.mail_count})
 
         return response
 
     def _connection_ssl(self):
-        connection = imaplib.IMAP4_SSL(self.server, self.port)
+        connection = imaplib.IMAP4_SSL(self.server, int(self.port))
         return connection
 
     def _connection_starttls(self):
-        connection = imaplib.IMAP4(self.server, self.port)
+        connection = imaplib.IMAP4(self.server, int(self.port))
         connection.starttls(create_default_context())
         return connection
 
+    def _disconnect(self):
+        try:
+            if self.connection is not None:
+                self.connection.close()
+                self.connection.logout()
+        finally:
+            self.connection = None
+
     def _get_mail_count(self):
         try:
-            mail_count = 0
-            directories = self.mailbox.split(',')
-
-            if self.security == "ssl":
-                connection = self._connection_ssl()
-            elif self.security == "starttls":
-                connection = self._connection_starttls()
-
-            connection.login(self.user, self.password)
-
-            for directory in directories:
-                connection.select(directory)
-                unseen_response = connection.search(None, self.criterion)
-                mails = unseen_response[1][0].split()
-                mail_count += len(mails)
-
-            connection.close()
-            return mail_count
-        except:
-            return None
+            while True:
+                if self.connection is None:
+                    if self.security == "ssl":
+                        self.connection = self._connection_ssl()
+                    elif self.security == "starttls":
+                        self.connection = self._connection_starttls()
+
+                    self.connection.login(self.user, self.password)
+
+                self.mail_count = 0
+                directories = self.mailbox.split(',')
+
+                for directory in directories:
+                    self.connection.select(directory)
+                    unseen_response = self.connection.search(None, self.criterion)
+                    mails = unseen_response[1][0].split()
+                    self.mail_count += len(mails)
+
+                if self.use_idle:
+                    self.connection.idle()
+                else:
+                    return
+        except socket_error:
+            self.mail_error = "Socket error";
+            self.connection = None
+            self.mail_count = None
+        except imaplib.IMAP4.error:
+            self.mail_error = "IMAP error";
+            self._disconnect()
```

Comment:
Explain this error. I received it here. If it's something trivial like occasional disconnection, we should keep going. Hard stop for bad settings only.

Reply:
this is 'everything smtp', docs don't do into much detail. it triggers on auth failure from my testing.